lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecomposition.cpp

```diff
@@ -357,6 +357,152 @@ struct ConvolutionToConv2dPattern
                              adaptor.getOperandConstraints());
 
     rewriter.replaceOp(op, output);
+    return success();
+  }
+};
+
+//===----------------------------------------------------------------------===//
+// Gather Pattern Matching
+//===----------------------------------------------------------------------===//
+
+struct GatherToEmbeddingConversionPattern
+    : public OpConversionPattern<ttir::GatherOp> {
+  using OpConversionPattern<ttir::GatherOp>::OpConversionPattern;
+
+  /**
+   * Validates Gather op constraints for embedding conversion
+   *
+   * Enforces constraints on Gather operation to ensure valid embedding
+   * transformation:
+   * - Output tensor shape: Multi-dimensional with last dimension as embedding
+   * size/ hiddenDim
+   * - Slice sizes: Must be [1, hiddenDim], where hiddenDim matches last output
+   * dimension
+   * - Offset dimensions: Strictly [2]
+   * - Collapsed slice dimensions: Strictly [0]
+   * - Start indices shape: Must be compatible with output shape, allowing
```

Comment:
Thanks for adding the comment. 
Can you just add bit more on the reshape? Why and when is it used? Probably the best way to explain it would be to start with how GatherOp works w.r.t. start indices, and how that translates to our EmbeddingOp. I want this to be very clear to any reader.